gui/src/3D/controls/holographicBackplate.ts

```diff
@@ -0,0 +1,140 @@
+    public get renderingGroupId(): number {
+        return this._model.renderingGroupId;
+    }
+
+    /**
+     * Gets the material used by the backplate
+     */
+    public get material(): FluentBackplateMaterial {
+        return this._material;
+    }
+
+    /**
+     * Gets a boolean indicating if this backplate shares its material with other HolographicBackplates
+     */
+    public get shareMaterials(): boolean {
+        return this._shareMaterials;
+    }
+
+    /**
+     * Creates a new holographic backplate
+     * @param name defines the control name
+     */
+    constructor(name?: string, shareMaterials = true) {
+        super(name);
+
+        this._shareMaterials = shareMaterials;
+    }
+
+    protected _getTypeName(): string {
+        return "HolographicBackplate";
+    }
+
+    private _rebuildContent(): void {
+
+    }
+
+    // Mesh association
+    protected _createNode(scene: Scene): TransformNode {
+        const collisionMesh = BoxBuilder.CreateBox((this.name ?? "HolographicBackplate") + "_CollisionMesh", {
+            width: 1.0,
+            height: 1.0,
+            depth: 1.0,
+        }, scene);
+        collisionMesh.isPickable = true;
+        collisionMesh.visibility = 0;
+
+        SceneLoader.ImportMeshAsync(
+            undefined,
+            HolographicBackplate.MODEL_BASE_URL,
+            HolographicBackplate.MODEL_FILENAME,
+            scene)
+            .then((result) => {
+                var importedModel = result.meshes[1];
+                importedModel.name = `${this.name}_frontPlate`;
+                importedModel.isPickable = false;
+                importedModel.parent = collisionMesh;
+                if (!!this._material) {
+                    importedModel.material = this._material;
+                }
+                this._model = importedModel;
+            });
+
+        return collisionMesh;
+    }
+
+    private _createMaterial(mesh: Mesh) {
+        this._material = new FluentBackplateMaterial(this.name + " Material", mesh.getScene());
+    }
+
+    protected _affectMaterial(mesh: Mesh) {
+        // Back
+        if (this._shareMaterials) {
+            if (!this._host._touchSharedMaterials["fluentBackplateMaterial"]) {
+                this._createMaterial(mesh);
+                this._host._touchSharedMaterials["fluentBackplateMaterial"] = this._material;
+            } else {
+                this._material = this._host._touchSharedMaterials["fluentBackplateMaterial"] as FluentBackplateMaterial;
+            }
+        } else {
+            this._createMaterial(mesh);
+        }
+
+        this._rebuildContent();
+    }
+
+    /**
+     * Releases all associated resources
+     */
+    public dispose() {
+        super.dispose(); // will dispose main mesh ie. back plate
+
+        this.onPointerMoveObservable.remove(this._pointerHoverObserver);
+
+        if (!this.shareMaterials) {
+            this._material.dispose();
+
+            if (this._pickedPointObserver) {

```

Comment:
No need for this inspection. It won't fail if it is null.

Also -  should it be inside this if loop? or outside of it?